### DISCUSSION TITLE
Styling the switch component

### DISCUSSION THREAD

**Discussion by vishnup-ditto** [mantinedev/mantine#5449] [Q&A]
How do we style the Switch component, the switch component as far as I understand works using `input type="checkbox"` html element. The checked prop doesn't change when I check or uncheck the component. How do I go about accessing the checked attribute and go ahhead styling it? An example of this would be helpful. Please see the video below to see how the `checked` attribute never changes. E.g taken from the mantine docs. 
![switch](https://github.com/mantinedev/mantine/assets/113330036/63756d8e-b20c-426c-8170-2b74b9a16d6e)

I want to style the checked state and non-checked state differently via CSS? How do I go about this?

**Comment by rtivital** [comment]
You can find Switch source code here – https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Switch/Switch.module.css
You can use the same selectors as in the source file to apply styles.

**Reply by vishnup-ditto** [reply]
Understood, had a look at it. Can you briefly explain why the checked attribute never changes here though? What is the idea behind that?

**Reply by rtivital** [reply]
It changes, otherwise the styling would not work.

**Reply by vishnup-ditto** [reply]
Please see the attached gif. Feels like something weird is going on.

**Reply by vishnup-ditto** [reply]
Understood the problem, the css pseudoclass of `:checked` is different from `checked` attribute

**Reply by kschedler** [reply]
I am looking for the same solution, to change the color of the track to a specific hex value when the switch is "on", but I am unable to figure out the CSS incantations to do it.  Could you share how you made this work?  Thank you!

**Comment by Akronae** [comment]
You can use `has` selector for that:
```css
  .mantine-Switch-root {
    &:has(input:not(:checked)) {
      .mantine-Switch-track {
        background-color: var(--mantine-color-orange-3);
      }
    }
  }
```
